### PR TITLE
Etcd cert generator

### PR DESCRIPTION
**Pull request by rithujohn191**
Importing the `kube-client-agent` and `kube-signer-server` from [coreos-inc/etcd-certificate-generator](https://github.com/coreos-inc/etcd-certificate-generator).

Summary of changes:
1. Add etcd client agent component. (already reviewed in coreos-inc repo)
2. Add etcd server signer component. (already reviewed in coreos-inc repo)
3. Modify Makefile to include new components.
4. Add docs about etcd cert generation. (already reviewed in coreos-inc repo)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -18,12 +18,34 @@ check:
...
+push/kube-signer-server: image/kube-signer-server
+	@docker push quay.io/coreos/kube-signer-server:$(IMAGE_TAG)
+
+image/kube-client-agent:
+	@docker build -t quay.io/coreos/kube-client-agent:$(IMAGE_TAG) -f $(ROOT_DIR)/dockerfiles/Dockerfile.kube-client-agent .
+
+push/kube-client-agent: image/kube-client-agent
+	@docker push quay.io/coreos/kube-client-agent:$(IMAGE_TAG)
+
+test:
```
Comment (abhinavdahiya) [inline Makefile:45]: `make check` already runs `go test`

If we do want sep target `go test -v $(shell go list ./... | grep -v '/e2e')` should be sufficient here.
Why -i ?

Comment (rithujohn191) [reply Makefile:45]: This is take from [dex Makefile](https://github.com/coreos/dex/blob/master/Makefile).

I added this so you can run just the unit tests separately from everything else running in make check.

whats the reasoning for using  grep -v '/e2e' vs grep -v '/vendor/'

**Review by abhinavdahiya — APPROVED** [review]
LGTM

File: Makefile (hunk 1 above)
Comment (abhinavdahiya) [reply Makefile:45]: I think go default tool-chain commands skips `vendor/` dir when running `./...`

**Comment by ericchiang** [comment]
Running the tests generates a file

```
$ make test
# ...
$ git status
On branch etcd-cert-generator
Your branch is up-to-date with 'rithujohn191/etcd-cert-generator'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)

        pkg/certagent/system:etcd-peer:test.key

nothing added to commit but untracked files present (use "git add" to track)
```

**Comment by ericchiang** [comment]
From IRL conversation, we should also rename `kube-signer-server` to `kube-etcd-signer-server`

**Comment by rithujohn191** [comment]
Renamed `kube-signer-server` to `kube-etcd-signer-server` and modified test to delete generated key file

**Review by ericchiang — APPROVED** [review]
lgtm, merge away

**Comment by rithujohn191** [comment]
FYI, I also opened issue https://github.com/coreos/kubecsr/issues/12 as a follow up of our conversation

**Pull request merged**